nucliadb/src/nucliadb/ingest/orm/knowledgebox.py

```diff
@@ -69,52 +72,23 @@ def __init__(self, txn: Transaction, storage: Storage, kbid: str):
         self.kbid = kbid
         self._config: Optional[KnowledgeBoxConfig] = None
 
-    async def get_config(self) -> Optional[KnowledgeBoxConfig]:
-        if self._config is None:
-            async with datamanagers.with_ro_transaction() as txn:
-                config = await datamanagers.kb.get_config(txn, kbid=self.kbid)
-            if config is not None:
-                self._config = config
-                return config
-            else:
-                return None
-        else:
-            return self._config
-
-    @classmethod
-    async def delete_kb(cls, txn: Transaction, kbid: str):
-        # Mark storage to be deleted
-        # Mark keys to be deleted
-        kb_config = await datamanagers.kb.get_config(txn, kbid=kbid)
-        if kb_config is None:
-            # consider KB as deleted
-            return
-        slug = kb_config.slug
-
-        # Delete main anchor
-        async with txn.driver.transaction() as subtxn:
-            key_match = datamanagers.kb.KB_SLUGS.format(slug=slug)
-            await subtxn.delete(key_match)
-
-            when = datetime.now().isoformat()
-            await subtxn.set(KB_TO_DELETE.format(kbid=kbid), when.encode())
-            await subtxn.commit()
-
-        audit_util = get_audit()
-        if audit_util is not None:
-            await audit_util.delete_kb(kbid)
-        return kbid
+    @staticmethod
+    def new_unique_kbid() -> str:
+        return str(uuid4())
 
     @classmethod
+    @processor_observer.wrap({"type": "create_kb"})
     async def create(
         cls,
         txn: Transaction,
         slug: str,
         semantic_model: SemanticModelMetadata,
         uuid: Optional[str] = None,
         config: Optional[KnowledgeBoxConfig] = None,
-        release_channel: ReleaseChannel.ValueType = ReleaseChannel.STABLE,
+        release_channel: Optional[ReleaseChannel.ValueType] = ReleaseChannel.STABLE,
```

Comment:
Optional is not needed when there is a default value no?

Reply:
It was to avoid conditionals on `KnowledgeBox.create` callers if they may not have the `release_channel`, but I can move the condition to the POST /kbs endpoint